src/test/javascript/spec/component/modeling-explanation-editor/modeling-explanation-editor.component.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { ModelingExplanationEditorComponent } from 'app/exercises/modeling/shared/modeling-explanation-editor.component';
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('ModelingExplanationEditorComponent', () => {
+    let fixture: ComponentFixture<ModelingExplanationEditorComponent>;
+    let comp: ModelingExplanationEditorComponent;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [FormsModule],
+            declarations: [ModelingExplanationEditorComponent],
+            schemas: [],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ModelingExplanationEditorComponent);
+                comp = fixture.componentInstance;
+            });
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    it('should initialize', () => {
+        fixture.detectChanges();
+        expect(ModelingExplanationEditorComponent).to.be.ok;
+    });
+
+    it('should change explanation value bidirectionally between component and template', () => {
+        comp.explanation = 'Initial Explanation';
+        fixture.detectChanges();
+        fixture.whenStable().then(() => {
```

Comment:
Why is this necessary? I thought `detectChanges` already takes care of "stabilising" the component.

Reply:
Without it textarea value does not get updated, I believe it is related to FormsModule.

Reply:
Ah ok.